### ISSUE TITLE
Ignoring unknown directives.

### ISSUE THREAD

**Issue by Y-Less** [pawn-lang/compiler#426]
In short, this should be allowed:

```pawn
#if 0
	#unknown_directive
#endif
```

Currently that gives an error, when it should be skipped.  Implementing this is trivially easy - I already did it:

https://github.com/Y-Less/compiler/commit/267c8f8ffdb250ba855aefda6b52f84d5d8fac0a

But there's one HUGE issue - what happens if that unknown and ignored directive from a future compiler version actually ends or somehow changes the `#if`?  E.g:

```pawn
#if 0
	printf("never");
#fi
```

An older compiler that sees that will just ignore `#fi`, not end the `#if`, and probably reach the end of the file without doing anything.  There are two solutions:

1. Ignore this problem.  If you want to make your code backwards-compatible just don't rely on older compilers correctly ignoring (or not) important directives.

2. Enumerate all possible future control-flow directives there ever may be, this could be, but is not limited to:

```pawn
#if
#else
#elseif
#elif
#ifdef
#elifdef
#elseifdef
#endif
#fi
#el
#ifndef
#elseifndef
#elifndef
```

Will we ever want `#iff`?  `#switch`, `#esac` etc?  We can't possibly know.

Thus I think option 1 is fine - just don't be an idiot!

**Comment by Zeex** [comment]
This was implemented in #427, so I'm closing this issue.

**Comment by Y-Less** [comment]
I forgot I did that!
